File: 23-03-2026/no_consecutive_repeats.js
function hasNoRepeats(str) {
  let noRepeats = true;

  for (let i = 0; i < str.length; i++) {
    if (str[i] == str[i + 1]) {
      noRepeats = false;
      break;
    }
  }

  return noRepeats;
}

console.log(hasNoRepeats("hi world")); // true.
console.log(hasNoRepeats("hello world")); // false.
console.log(hasNoRepeats("abcdefghijklmnopqrstuvwxyz")); // true.
console.log(hasNoRepeats("freeCodeCamp")); // false.
console.log(hasNoRepeats("The quick brown fox jumped over the lazy dog.")); // true.
console.log(hasNoRepeats("Mississippi")); // false.

File: 23-03-2026/no_consecutive_repeats.py
def has_no_repeats(s):
    noRepeats, sLength = True, len(s);

    for i in range(sLength):
        if i + 1 < sLength:
            if s[i] == s[i + 1]:
                noRepeats = False;
                break;

    return noRepeats;

print(has_no_repeats("hi world")); # True.
print(has_no_repeats("hello world")); # False.
print(has_no_repeats("abcdefghijklmnopqrstuvwxyz")); # True.
print(has_no_repeats("freeCodeCamp")); # False.
print(has_no_repeats("The quick brown fox jumped over the lazy dog.")); # True.
print(has_no_repeats("Mississippi")); # False.
